### ISSUE TITLE
Licensing

### ISSUE THREAD

**Issue by filippo82** [softwareunderground/subsurface-journal#2]
I see that most OA journals require authors to accept the [**CC-BY**](https://creativecommons.org/licenses/by/4.0/) license.

Is this going to be a good choice for the Journal too?

**Comment by JesperDramsch** [comment]
Short answer: Yes, except for software.

Long answer: We could do it like EarthArxiv and provide a choice of 
- CC-BY (Creative Commons)
- LGPL (Lesser Gnu Public License)
- CC0 (Copyleft)
- AFL (Academic Free License)

http://help.osf.io/m/preprints/l/726873-preprint-faqs#how_should_i_license_my_preprint

However, they provide these to comply with journals. As a journal, we have "the power" to choose how we want to license. CC-BY is a very nice license, as it's open, permissive and requires attribution. It's my fave

For software submissions, we should consider a software license, however. Here MIT, GPL, Apache or BSD are choices. [More info](https://choosealicense.com/)

**Comment by filippo82** [comment]
Yes @JesperDramsch, paper and software licenses are two different topics!

For the papers, I would prefer to stick to one license and then, for the software, offers a choice (as you suggested.

**Comment by prisae** [comment]
> For software submissions, we should consider a software license, however. Here MIT, GPL, Apache or BSD are choices.

Crucial point. And the four you mentioned should be made available. Not just GPL or just MIT, but give the author a choice but also provide links where they can decide what to use, because still many are not aware of the differences.

**Comment by leouieda** [comment]
I don't really like the EarthArxiv options because I've seen people licensing the paper itself with LGPL, which doesn't make much sense. 

For **paper license**, we could say "papers are CC-BY" and I'd be OK with that. Most people don't know enough to be able to make a conscious choice.

For software, I don't think they should be a part of the submission. We shouldn't be hosting the software archive itself. Instead, we should require authors to archive the software/data on Zenodo/figshare/etc and we include that information in the article. This is what JOSS does:

![2019-03-08-134925_1920x1080_scrot](https://user-images.githubusercontent.com/290082/54062282-340c2380-41a9-11e9-86a9-3d78b322bf2a.png)

Making the archive is the last step before publication and only done after acceptance.

For the **software license**, I don't see why we need to restrict the options. If it's OSI approved, then it should be acceptable to us. But this should be done in the software archive, not in the journal. Having an OSI approved license for the code should be a required step for publication.

**Comment by JesperDramsch** [comment]
> For **paper license**, we could say "papers are CC-BY" and I'd be OK with that. Most people don't know enough to be able to make a conscious choice.

Agreed

> We shouldn't be hosting the software archive itself. Instead, we should require authors to archive the software/data on Zenodo/figshare/etc and we include that information in the article. This is what JOSS does:
> Making the archive is the last step before publication and only done after acceptance.

I think we're down to discussing semantics and it's probably down to me not being able to articulate it properly, sorry. I agree that the hosting should be one of the acceptable places like zenodo. I think code quality should be part of the review process (similar to JOSS). (So running the code and checking the outputs / tests.) So I guess it doesn't have to be "submitted" but it has to be available, testable and reproducible and for acceptance also archived. (I am guessing we probably agree here actually. So it's probably a long-winded self-indulgent way of saying: I agree.)
 
> For the **software license**, I don't see why we need to restrict the options. If it's OSI approved, then it should be acceptable to us. But this should be done in the software archive, not in the journal. Having an OSI approved license for the code should be a required step for publication.

Good point. It's also somewhat "future-proofing" and keeping us flexible without excluding wanted submissions.

**Comment by leouieda** [comment]
> I am guessing we probably agree here actually. So it's probably a long-winded self-indulgent way of saying: I agree.

:rofl: Sorry if it seemed like I disagreed with you. That was just my attempt at distilling what I thought you had in mind. Apparently it was successful ;)

**Comment by JesperDramsch** [comment]
Ha! It's the GitHub issue equivalent of missing a high five!

**Comment by filippo82** [comment]
@JesperDramsch @leouieda good discussion!

It seems we are converging to something like:

- Paper license: CC-BY
- Software license: it has to be OSI approved
- Software is not submitted together with the manuscript but it **has** to be available on GitHub (or similar).
- Final version of the software (corresponding to the accepted manuscript) shall be hosted on Zenodo (or similar).

I will ask a question about software on the #1 issue.
